### ISSUE TITLE
Permission denied on auth/token/renew-self

### ISSUE THREAD

**Issue by shayangz** [hashicorp/vault#3247]
This is on Vault `0.8.1`:

I have a renewable valid token with `default` policy attached but it gets `permission denied` when trying to renew itself.

Below is the `lookup-self` response

```
$ curl -s --header "X-Vault-Token: $VT" https://vault.service.consul:8200/v1/auth/token/lookup-self | jq .
{
  "request_id": "4edaa8bd-10bb-ff00-2e01-8d7df7c69cc6",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": {
    "accessor": "22f38afa-d825-952c-afa3-cda05f7d3e70",
    "creation_time": 1503889319,
    "creation_ttl": 21600,
    "display_name": "aws-node",
    "expire_time": "2017-08-29T03:01:59.630922845Z",
    "explicit_max_ttl": 0,
    "id": "XXXX",
    "issue_time": "2017-08-28T03:01:59.630913981Z",
    "last_renewal": "2017-08-28T15:02:34.473193306Z",
    "last_renewal_time": 1503932554,
    "meta": {
      "account_id": "XXXX",
      "auth_type": "iam",
      "canonical_arn": "arn:aws:iam::...",
      "client_arn": "arn:aws:sts::...",
      "client_user_id": "XXXX",
      "inferred_aws_region": "us-west-2",
      "inferred_entity_id": "i-XXXX",
      "inferred_entity_type": "ec2_instance"
    },
    "num_uses": 0,
    "orphan": true,
    "path": "auth/aws/login",
    "policies": [
      "default",
      "lr-all"
    ],
    "renewable": true,
    "ttl": 34559
  },
  "wrap_info": null,
  "warnings": null,
  "auth": null
}
```

but when I do 

```
$ curl -s --header "X-Vault-Token: $VT" https://vault.service.consul:8200/v1/auth/token/renew-self
{
  "errors": [
    "permission denied"
  ]
}
```

The token was generated using the `auth/aws/login` endpoint using an AIM method (with a `period` value of 6h).

any clues?

**Comment by jefferai** [comment]
What is the output of `vault policies default` and `vault policies lr-all`?

**Comment by shayangz** [comment]
We haven't modified `default`:

```
$ vault policies default

# Allow tokens to look up their own properties
path "auth/token/lookup-self" {
    capabilities = ["read"]
}

# Allow tokens to renew themselves
path "auth/token/renew-self" {
    capabilities = ["update"]
}

# Allow tokens to revoke themselves
path "auth/token/revoke-self" {
    capabilities = ["update"]
}

# Allow a token to look up its own capabilities on a path
path "sys/capabilities-self" {
    capabilities = ["update"]
}

# Allow a token to renew a lease via lease_id in the request body; old path for
# old clients, new path for newer
path "sys/renew" {
    capabilities = ["update"]
}
path "sys/leases/renew" {
    capabilities = ["update"]
}

# Allow looking up lease properties. This requires knowing the lease ID ahead
# of time and does not divulge any sensitive information.
path "sys/leases/lookup" {
    capabilities = ["update"]
}

# Allow a token to manage its own cubbyhole
path "cubbyhole/*" {
    capabilities = ["create", "read", "update", "delete", "list"]
}

# Allow a token to wrap arbitrary values in a response-wrapping token
path "sys/wrapping/wrap" {
    capabilities = ["update"]
}

# Allow a token to look up the creation time and TTL of a given
# response-wrapping token
path "sys/wrapping/lookup" {
    capabilities = ["update"]
}

# Allow a token to unwrap a response-wrapping token. This is a convenience to
# avoid client token swapping since this is also part of the response wrapping
# policy.
path "sys/wrapping/unwrap" {
    capabilities = ["update"]
}
```

and `lr-all` is 

```
path "lr_infra_pki_root2017/*" {
  policy = "deny"
}

path "lr_infra_pki_root2017/cert/ca" {
  policy = "read"
}

path "lr_infra_pki_intermediate/*" {
  policy = "deny"
}

path "lr_infra_pki_intermediate/issue/consul_service" {
  capabilities = ["create", "update"]
}

path "lr_infra_pki_intermediate/issue/consul_template_service" {
  capabilities = ["create", "update"]
}

path "secret/sensu_rabbitmq" {
  capabilities = ["read"]
}
```

**Comment by jefferai** [comment]
Oh, sorry. Sometimes it's easy to miss the little things. The issue is that curl defaults to GET but it's an 'update' operation so you need to use POST. Add -X POST to curl and you should be good!

**Comment by shayangz** [comment]
🤦‍♂️ 

quick question: the response of self-lookup shows `expire_time` 24 hours after `issue_time` for the token. Will this token not work past the `expire_time` regardless of renewals? How can we make it such that it never expires as long as it's being renewed within the specific `period` (6h in this case)?

**Comment by jefferai** [comment]
The expire time is calculated based on the current TTL of the token. Periodic tokens (generally) don't expire so long as you renew them within the period. There is a bug with AWS IAM periodic token support where this isn't true, but will be fixed in an upcoming 0.8.2.
